net/fetcher.go

```diff
@@ -13,28 +13,60 @@ import (
 	"github.com/filecoin-project/go-filecoin/types"
 )
 
-// Fetcher is used to fetch data over the network.  It is implemented with
+// Fetcher defines an interface that may be used to fetch data from the network.
+type Fetcher interface {
```

Comment:
The syncFetcher interface is defined in the consumer as I would expect.  What does this guy do?  It appears extraneous.

Reply:
Its used by the node: https://github.com/filecoin-project/go-filecoin/pull/3047/files#diff-380e79adccbaf1276bf6c91cc551cd7fR161 and I think I might remove the `syncFetcher` interface as it is no different than the `Fetcher` interface.

Reply:
I think the consumer side interface definition is helpful when asking for a subset of functions from something big, like porcelain and plumbing. But I don't think its a pattern we should use everywhere. As an example I am having some mixed feelings about the number of `*ChainReader`/`*ChainProvider` interfaces we have defined (at least 10) -- feels like we might want to rethink this strategy and create 2  or 3 "cancolical" chain interface (read, write, query) that we use everywhere with corresponding Mocks/Fakes used in testing. My 2 cents here at least.

Reply:
I agree with that proposal.